File: app/src/main/java/org/schabi/newpipe/fragments/BackPressable.java
package org.schabi.newpipe.fragments;

/**
 * Indicates that the current fragment can handle back presses.
 */
public interface BackPressable {
    /**
     * A back press was delegated to this fragment.
     *
     * @return if the back press was handled
     */
    boolean onBackPressed();
}

File: app/src/main/java/org/schabi/newpipe/fragments/BackPressable.kt
package org.schabi.newpipe.fragments

/**
 * Indicates that the current fragment can handle back presses.
 */
interface BackPressable {
    /**
     * A back press was delegated to this fragment.
     *
     * @return if the back press was handled
     */
    fun onBackPressed(): Boolean
}
